### ISSUE TITLE
Infinite loop because constraintlayout use int and maui use double

### ISSUE THREAD

**Issue by xtuzy** [xtuzy/SharpConstraintLayout#23]
Problem:
i update maui example in f2d46a3, a grid split to two part, one part is a constraintlayout, but constraintlayout can't deal with it, because if grid split give a not int value, constraintlayout alway return int, that let layout will infinite loop because result have bias.

For solve this problem, i would try use px as unit when measure and layout in constraintlayout, but set property and fluentlayoytconstraintset use dp, return size to maui a dp value, that will let it not always is int.

**Comment by xtuzy** [comment]
初步在dptopx分支修改后，依然无限循环。使用"*,*"时, grid传递给child的大小总是grid的大小，这导致constraintlayout也用这个大小去布局，我觉得传下来的高度应该是grid的一半，看maui的issue，造成无限循环它也可能不是我的问题，见
https://github.com/dotnet/maui/issues/14694 ，
https://github.com/dotnet/maui/issues/14520

我需要把例子调回net6再运行下看是否是版本的问题

**Comment by xtuzy** [comment]
重新运行主分支的例子, 我看到Log中285.5 变成了285:
```
MeasureLayout View=ConstraintLayout Id=-1132297589 availableSize={Width=484 Height=285.5} Spec=(MeasureSpec: EXACTLY 484 x MeasureSpec: EXACTLY 285)
```


这是因为:
https://github.com/xtuzy/SharpConstraintLayout/blob/f2d46a39844b7f49d1d257c5afe89de8c415aea8/SharpConstraintLayout.Maui/Widget/ConstraintLayout.cs#L445
修改成:
```
int availableWidth = (int)(availableSize.Width + 0.5);
int availableHeight = (int)(availableSize.Height + 0.5);
```
Log中285.5 变成了286:
```
MeasureLayout View=ConstraintLayout Id=-1132297589 availableSize={Width=484 Height=285.5} Spec=(MeasureSpec: EXACTLY 484 x MeasureSpec: EXACTLY 286)
```
之后运行成功. 

fix in 5afb519bd81e4bb80e32fba388d5301cfe1bc1cd , 我还增加了点对其它doube to int的修改, 暂时不知道影响, 其它测试的外观都还正常
test in a816e75ab4db97107d7f479845d135f5b774f34f , FitGridTest的结果可以看到最终分配的大小还是按照Grid的设置分的, 所以对于ConstraintLayout来说, 内部会有1dp的偏差, 如果使用px, 偏差应该减小到1px, 但当前未打算修改

- [ ]  convert dp and px
